### DISCUSSION TITLE
How to run tests in separate process?

### DISCUSSION THREAD

**Discussion by bbrandao** [pestphp/pest#1008] [General]
Hi all,

Phpunit has some annotations like:
/**
 * @runTestsInSeparateProcesses
 * @runInSeparateProcess
 * @preserveGlobalState disabled
 */

To indicate that the test should run in separate process. This avoid some problems with mocks.

How I can do that in Pest?

Best regards,
Bruno

**Comment by owenvoke** [comment]
This isn't possible in Pest, and I don't think there are plans to add support (unless someone PRs it). 🤔

Here are some previous issues related to process isolation (https://github.com/pestphp/pest/issues/270 and https://github.com/pestphp/pest/issues/665) and a [PR that tried to implement it](https://github.com/pestphp/pest/pull/285).

If you'd like to work on this, I'm sure that it'd be welcome. 🙂

**Reply by bbrandao** [reply]
Thnaks for your reply.

Do you know how to solve a problem in PEST when we mock a class Mockery::mock('overload:' SOME_CLASS::class)?

**Reply by mreduar** [reply]
> Thnaks for your reply.
> 
> Do you know how to solve a problem in PEST when we mock a class Mockery::mock('overload:' SOME_CLASS::class)?

Were you able to find a solution to this problem at the time? I've been trying to solve the problem for hours

**Reply by bbrandao** [reply]
With Pest I can't solve this problem. In this project I came back to PHPUnit.
